src/main/angular/app/graph/graph.component.ts

```diff
@@ -41,23 +46,46 @@ export class GraphComponent implements OnInit {
 
   /** Whether to show the chart. When it's not selected, prompt the user to select a project. */
   public shouldShowChart: boolean;
+  /** The text that's shown when the chart isn't. */
+  public noChartMessage: string;
 
   constructor(
     private dataService: DataService,
     private graphProcessor: GraphProcessorService
   ) {
     this.shouldShowChart = false;
     this.projects = [];
+    this.noChartMessage = GraphComponent.LOADING_MESSAGE;
   }
 
   /** Called when an input field changes. */
-  ngOnChanges(changes: SimpleChanges) {
-    this.shouldShowChart = this.projects.length > 0;
-    this.graphProcessor.processChanges(
+  async ngOnChanges(changes: SimpleChanges) {
+    if (changes.projects.isFirstChange()) {
+      // We're still retrieving the list of projects
+      this.shouldShowChart = false;
+      this.noChartMessage = GraphComponent.LOADING_MESSAGE;
+    } else if (!this.projects || this.projects.length === 0) {
```

Comment:
is it ever the case that this.projects is null? If not, is this check unnecessary?

Reply:
Quite right! removed.